core/platform/lf_patmos_support.c

```diff
@@ -0,0 +1,89 @@
+#include <time.h>
+#include <errno.h>
+#include <assert.h>
+#include <lf_patmos_support.h>
+#include "../platform.h"
+#include <machine/rtc.h>
+#include <machine/exceptions.h>
+#include <stdio.h>
+
+/**
+ * @brief Sleep until an absolute time.
+ * TODO: For improved power consumption this should be implemented with a HW timer and interrupts.
+ *
+ * @param wakeup int64_t time of wakeup
+ * @return int 0 if successful sleep, -1 if awoken by async event
+ */
+
+int _lf_interruptable_sleep_until_locked(environment_t* env, instant_t wakeup) {
+    instant_t now;
+    _lf_async_event = false;
+    lf_disable_interrupts_nested();
```

Comment:
Here interrupts should be enabled, not disabled. I think this is copied from the arduino support? I found the same bug there